exonum-java-binding/core/rust/src/utils/resource_manager/imp.rs

```diff
@@ -57,12 +58,9 @@ fn add_handle_impl<T: 'static>(handle: Handle, ownership: HandleOwnershipType) {
     assert_ne!(handle, 0);
     assert!(
         HANDLES_MAP
-            .write()
-            .expect("Unable to obtain write-lock")
             .insert(handle, HandleInfo::new(TypeId::of::<T>(), ownership))
             .is_none(),
         "Trying to add the same handle for the second time: {:X}",
```

Comment:
Assuming `insert` returns the previous value, does there happen an *overwrite* of the previous value (i.e., the change of the state of the map in case of an invalid argument)?

Reply:
Re-written using CHashMap.alter()

Reply:
FYI, I bumped into another library, DashMap, it has proper [get_or_insert](https://docs.rs/dashmap/1.2.1/dashmap/struct.DashMap.html#method.get_or_insert) in its API.

Hovewer, even though it is 1.0+, there are no concurrent tests and even complete functional tests 👎